### ISSUE TITLE
[BUG Report] torch.from_numpy()

### ISSUE THREAD

**Issue by JimLee1996** [pytorch/pytorch#7079]
## Issue description

It may take a long time to run, sometimes even causes python kernel die when using function `torch.from_numpy()` to convert int32 nd.array to torch.Tensor.(float64 is not effected. that's weird)

## Code example

```python
import numpy as np
import torch

# normal 
a = np.random.randn(41, 288, 360, 3)
torch.from_numpy(a)

# gg, slow or dead
b = np.random.randint(10, size=(41, 288, 360, 3))
torch.from_numpy(b)
```

## System Info

- PyTorch
- pip
- Windows10 ltsb 2015 
- PyTorch version: 0.4.0
- Python version: 3.6.4
- CUDA/cuDNN version: not use
- GPU models and configuration: not use

## Additions

- I've also tested on Mac 10.11.6 and Ubuntu 16.04. The same problem occurs.

- lots of numpy images' dtype=uint8. It may have bad effects on program effeciency.

**Comment by DEMON3** [comment]
wired

**Comment by DEMON3** [comment]
weird

**Comment by Alok** [comment]
Getting this on macOS 10.13.4 too.

**Comment by fmassa** [comment]
Can't reproduce on master `'0.5.0a0+4caea64'` using Linux, compiled from source.

**Comment by JimLee1996** [comment]
## update

It's fine to assign,

```python
import numpy as np
import torch

a = np.random.randn(41, 288, 360, 3)
aa = torch.from_numpy(a)

b = np.random.randint(10, size=(41, 288, 360, 3))
bb= torch.from_numpy(b)
```

However, when printing

```python
# normal
print(aa)

# ......
print(bb)
```

Maybe it will have no effect on program performance.

**Comment by SsnL** [comment]
Can't reproduce on `'0.5.0a0+3b63be0'` using macos, compiled from source. I printed the tensors.

**Comment by fmassa** [comment]
So, my first thought was that https://github.com/pytorch/pytorch/issues/6863 was the culprit, but I think that it was merged into 0.4 before the release?

**Comment by SsnL** [comment]
I can actually reproduce on 0.4 with macos. And after inspecting the python scripts from conda binary... it seems that #6863 is not in 0.4.

**Comment by SsnL** [comment]
@fmassa I just checked. It isn't in 0.4 branch either

**Comment by JimLee1996** [comment]
I can print it either. But it takes much more time to print(bb) than print(aa).

- pytorch version is 0.4.0
- CPU: Xeon E5-2620 v4
- Win10 LTSB 2015

=====================

- a->aa, 0.0s
- b->bb, 0.0010232925415039062s
- print(aa), 0.8235607147216797s
- print(bb), **72.6184949874878s**

## test script
```python
import numpy as np
import torch
import time

a = np.random.randn(41, 288, 360, 3)
b = np.random.randint(10, size=(41, 288, 360, 3))

start = time.time()
aa = torch.from_numpy(a)
a_time = time.time() - start

start = time.time()
bb = torch.from_numpy(b)
b_time = time.time() - start

start = time.time()
print(aa)
aa_time = time.time() - start

start = time.time()
print(bb)
bb_time = time.time() - start

print(a_time, b_time, aa_time, bb_time)
```

**Comment by SsnL** [comment]
@JimLee1996 Yes, I know. When I say that I can print I meant that I can print it quickly. I just manually patched #6863 in 0.4 and verified that it works. So this is fixed.

**Comment by JimLee1996** [comment]
Can't reproduce on `0.4.1`. Fixed. Thank you.
